pkg/cli/builder/table/funcs.go

```diff
@@ -147,11 +154,38 @@ func cleanFields(obj any) any {
 			}
 		}
 		ro.SetAnnotations(annotations)
+
+		// decode secret values
+		if sec, ok := ro.(*apiv1.Secret); ok {
+			return decodeSecret(sec)
+		}
```

Comment:
We touched on this lightly offline, but I think we could also invert the dependency so that the type implementer can tell us how cleanup should work for their type; For example:

```go
package table
// ...

// Cleanable describes types that implement custom cleanup logic that should be called before writing them to a table buffer.
type Cleanable interface {
  // Clean returns a cleaned up copy of the underlying Cleanable.
  Clean() any
}
//...


func cleanFields(obj any) any {
  //...
  ro.SetAnnotations(annotations)

  // Finally, run any custom cleaning logic
  if c, ok := ro.(Cleanable); ok {
    return c.Clean()
  } 
  //...
}
```

Meanwhile, over at the type definition:

```go
package v1
//...

func (s *Secret) Clean() any {
  decodedSecret := struct {
    metav1.TypeMeta   `json:",inline"`
    metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
  
    Type string            `json:"type,omitempty"`
    Data map[string]string `json:"data,omitempty"`
	Keys []string          `json:"keys,omitempty"`
  }{
    TypeMeta:   s.TypeMeta,
    ObjectMeta: s.ObjectMeta,
	Type:       s.Type,
	Data:       map[string]string{},
	Keys:       s.Keys,
  }
  for k, v := range sec.Data {
    decodedSecret.Data[k] = string(v)
  }
  return decodedSecret
}

```

The benefit is that we don't need to perform surgery on the table package whenever we need to extend cleanup logic.

Reply:
I like that and now I understand what you wanted to tell me there :grin: 
Is it good to have that logic, which we only need for CLI output, in the API code though?
I guess it doesn't do any harm there, just wondering.

Reply:
> Is it good to have that logic, which we only need for CLI output, in the API code though?

Ah, that's a great point. Now I'm on the fence about the value of the suggestion.

I don't want to hold this fix up when we can always refactor later if we think of something nicer.